gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/encryption/EncryptionConfiguration.java

```diff
@@ -0,0 +1,162 @@
+
+    @Value("${management.mongodb.encryption.keyVault.collectionName:__dataKeys}")
+    private String keyVaultCollectionName;
+
+    @Value("${management.mongodb.prefix}")
+    private String tablePrefix;
+
+    private final Environment environment;
+
+    public EncryptionConfiguration(Environment environment) {
+        this.environment = environment;
+    }
+
+    private String getDatabaseName() {
+        String uri = environment.getProperty("management.mongodb.uri");
+        if (uri != null && !uri.isEmpty()) {
+            // Remove user:password from the URI as it can contain special characters and isn't needed for the database name
+            String uriWithoutCredentials = uri.replaceAll("://.*@", "://");
+            return URI.create(uriWithoutCredentials).getPath().substring(1);
+        }
+
+        return environment.getProperty("management.mongodb.dbname", "gravitee");
+    }
+
+    private Map<String, Map<String, Object>> loadKmsProvidersFromConfig() {
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<>();
+        boolean found = true;
+        int idx = 0;
+
+        while (found) {
+            String type = environment.getProperty(KEY_MANAGEMENT_SYSTEM_PROPERTY + "[" + idx + "].type");
+            found = (type != null);
+            if (found) {
+                switch (type) {
+                    case KEY_MANAGEMENT_SYSTEM_LOCAL_TYPE:
+                        kmsProviders.put(type, loadLocalKmsProviderFromConfig(idx));
+                        break;
+                    default:
+                        log.warn("Unsupported key management system type: {}", type);
+                        break;
+                }
+            }
+            idx++;
+        }
+        if (kmsProviders.isEmpty()) {
+            throw new IllegalStateException("No key management system provider found");
+        }
+        return kmsProviders;
+    }
+
+    private Map<String, Object> loadLocalKmsProviderFromConfig(int idx) {
+        Map<String, Object> localKmsProvider = new HashMap<>();
+        String masterKey = environment.getProperty(KEY_MANAGEMENT_SYSTEM_PROPERTY + "[" + idx + "].local.key");
+        if (masterKey == null) {
+            throw new IllegalStateException("Master key not set");
+        }
+        localKmsProvider.put(KEY_MANAGEMENT_SYSTEM_LOCAL_KEY_PROPERTY, masterKey);
+
+        return localKmsProvider;
+    }
+
+    @Bean
+    @Conditional(EncryptionEnabledCondition.class)
+    public ClientEncryption clientEncryption(MongoFactory mongoFactory) {
+        // Key Management System (KMS) providers.
+        Map<String, Map<String, Object>> kmsProviders = loadKmsProvidersFromConfig();
+
+        ClientEncryptionSettings.Builder builder = ClientEncryptionSettings
+            .builder()
+            // The collection in MongoDB where the Data Encryption Keys (DEKs) will be stored.
+            .keyVaultNamespace(getDatabaseName() + "." + tablePrefix + keyVaultCollectionName)
```

Comment:
We can try removing the getDatabaseName() and see if it reuses the same database from the mongoFactory.

Reply:
It works without the database name. I'll remove that part.